go/interpreter/lfvm/memory.go

```diff
@@ -224,7 +224,9 @@ func (m *Memory) CopyData(offset uint64, trg []byte) {
 	}
 }
 
-func (m *Memory) GetSlice(offset, size uint64) []byte {
+// readSlice reads size bytes from the memory at the given offset.
+// Returns nil if offset+size is out of bounds or size is 0.
+func (m *Memory) readSlice(offset, size uint64) []byte {
```

Comment:
I would refactor this, to enforce the difference between normal use of memory and extended introspection used by test/ct-adapter:

- During normal operation, any read must expand memory first. Having this method invites calling `readSlice` and developer has to think if expansion needs to be done or not, this is not needed. 
- During ct-adapter/testing, a method get rawMemory returning the complete memory can be used and using slice operators to scope it would make the job (or even accessing internal slice directly) 

Factoring this code inside `readSliceAndExpandMemory` would remove the need for these checks. 
- size==0 yields empty slice
- mandatory expansion removes the need to check size.

Reply:
Sorry, I did not propose any renaming. Because the returned slice is a reference to the mem storage, I guess it is a better name than readSlice